server/base/src/main/java/org/apache/accumulo/server/metadata/ServerAmpleImpl.java

```diff
@@ -138,11 +143,66 @@ public void putGcFileAndDirCandidates(TableId tableId, Collection<ReferenceFile>
       return;
     }
 
-    try (BatchWriter writer = createWriter(tableId)) {
+    try (BatchWriter writer = context.createBatchWriter(DataLevel.of(tableId).metaTable())) {
       for (var fileOrDir : candidates) {
         writer.addMutation(createDeleteMutation(fileOrDir));
       }
-    } catch (MutationsRejectedException e) {
+    } catch (MutationsRejectedException | TableNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void addBulkLoadInProgressFlag(String path, long fateTxid) {
+
+    // Bulk Import operations are not supported on the metadata table, so no entries will ever be
+    // required on the root table.
+    Mutation m = new Mutation(BlipSection.getRowPrefix() + path);
+    m.put(EMPTY_TEXT, EMPTY_TEXT, new Value(FateTxId.formatTid(fateTxid)));
+
+    try (BatchWriter bw = context.createBatchWriter(MetadataTable.NAME)) {
+      bw.addMutation(m);
+    } catch (MutationsRejectedException | TableNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeBulkLoadInProgressFlag(String path) {
+
+    // Bulk Import operations are not supported on the metadata table, so no entries will ever be
+    // required on the root table.
+    Mutation m = new Mutation(BlipSection.getRowPrefix() + path);
+    m.putDelete(EMPTY_TEXT, EMPTY_TEXT);
+
+    try (BatchWriter bw = context.createBatchWriter(MetadataTable.NAME)) {
+      bw.addMutation(m);
+    } catch (MutationsRejectedException | TableNotFoundException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void removeBulkLoadEntries(TableId tableId, long tid) {
```

Comment:
Could add validation that the table id is a user table.

```suggestion
  public void removeBulkLoadEntries(TableId tableId, long tid) {
      Preconditions.checkArgument(DataLevel.of(tableId) == DataLevel.USER);
```

Reply:
`DataLevel.of()`, only compares tableId against the `root` and `metadata` table names and returns `user` by default.
https://github.com/apache/accumulo/blob/c8cd5809f9d4e3dcecce3d330b71b7ca794723c5/core/src/main/java/org/apache/accumulo/core/metadata/schema/Ample.java#L107-L116 

Since `removeBulkLoadEntries` only scans the metadata table, adding the proposed check would catch situations where `metadata` and `root` could be passed accidentally but it would allow invalid user table names. 

Do we want to allow that vs fast failing this operation?

Reply:
> but it would allow invalid user table names.

We could possibly add validation of table id to the table id code in another PR. User tables ids are base 36 numbers and system tables are ids a known set, so should be able to validate.  Our tests may be problematic if they manually generate incorrect tables ids.  

The code that generates a new user table id is here.

https://github.com/apache/accumulo/blob/c8cd5809f9d4e3dcecce3d330b71b7ca794723c5/server/manager/src/main/java/org/apache/accumulo/manager/tableOps/create/CreateTable.java#L78